src/MySqlConnector/Core/SchemaProvider.cs

```diff
@@ -38,24 +39,99 @@ public SchemaProvider(MySqlConnection connection)
+			{ nameof(FillTableConstraints), new[] { "TABLE_CATALOG", "TABLE_SCHEMA", "TABLE_NAME", "COLUMN_NAME" } },
+			{ nameof(FillIndexColumns), new[] { "TABLE_CATALOG", "TABLE_SCHEMA", "TABLE_NAME", "COLUMN_NAME" } },
+			{ nameof(FillViews), new[] { "TABLE_CATALOG", "TABLE_SCHEMA", "TABLE_NAME" } },
+			{ nameof(FillTables), new[] { "TABLE_CATALOG", "TABLE_SCHEMA", "TABLE_NAME" } },
+			{ nameof(FillDatabases), new[] { "SCHEMA_NAME" } },
+
+			// propertly some restrictions are missing
+		};
+	}
+
+	private async Task FillForeignKeys(IOBehavior ioBehavior, DataTable dataTable, string?[] restrictionValues, CancellationToken cancellationToken)
+	{
+		dataTable.Columns.AddRange(new DataColumn[]
+		{
+			new("CONSTRAINT_CATALOG", typeof(string)),
+			new("CONSTRAINT_SCHEMA", typeof(string)),
+			new("CONSTRAINT_NAME", typeof(string)),
+			new("TABLE_CATALOG", typeof(string)),
+			new("TABLE_SCHEMA", typeof(string)),
+			new("TABLE_NAME", typeof(string)),
+			new("COLUMN_NAME", typeof(string)),
+		});
+
+		var restrictionFilters = CreateRestrictionFilter(nameof(FillForeignKeys), dataTable, restrictionValues);
+
+		await FillDataTableAsync(ioBehavior, dataTable, "KEY_COLUMN_USAGE", restrictionFilters, cancellationToken).ConfigureAwait(false);
 	}
 
-	public ValueTask<DataTable> GetSchemaAsync(IOBehavior ioBehavior, CancellationToken cancellationToken) => GetSchemaAsync(ioBehavior, "MetaDataCollections", cancellationToken);
+	private List<string> CreateRestrictionFilter(string mappingKey, DataTable dataTable, string?[] restrictionValues)
+	{
+		List<string> restrictionFilters = new List<string>();
+		if (m_restrictionMapping.ContainsKey(mappingKey))
+		{
+			var restrictionMapping = m_restrictionMapping[mappingKey];
+			for (int i = 0; i < restrictionValues.Length; i++)
+			{
+				if (!string.IsNullOrEmpty(restrictionValues[i]))
+				{
+					if (i >= restrictionMapping.Length || !dataTable.Columns.Contains(restrictionMapping[i]))
+						throw new InvalidOperationException("Wrong restriction value mapping detected.");
+
+					restrictionFilters.Add($"UPPER({restrictionMapping[i]}) = '{restrictionValues[i]?.ToUpperInvariant()}'");
+				}
+			}
+		}
+		return restrictionFilters;
+	}
 
-	public async ValueTask<DataTable> GetSchemaAsync(IOBehavior ioBehavior, string collectionName, CancellationToken cancellationToken)
+	private async Task FillIndexColumns(IOBehavior ioBehavior, DataTable dataTable, string?[] restrictionValues, CancellationToken cancellationToken)
+	{
+		dataTable.Columns.AddRange(new DataColumn[]
+		{
+			new("TABLE_CATALOG", typeof(string)),
+			new("TABLE_SCHEMA", typeof(string)),
+			new("TABLE_NAME", typeof(string)),
+			new("NON_UNIQUE", typeof(string)),
+			new("INDEX_SCHEMA", typeof(string)),
+			new("INDEX_NAME", typeof(string)),
+			new("COLUMN_NAME", typeof(string)),
+			new("COLLATION", typeof(string)),
+			new("CARDINALITY", typeof(string)),
```

Comment:
MySql.Data returns the following columns from its `IndexColumns` schema:

```
INDEX_CATALOG
INDEX_SCHEMA
INDEX_NAME
TABLE_NAME
COLUMN_NAME
ORDINAL_POSITION
SORT_ORDER
```

Is there a reason to be different here?